funcx_websocket_service/application.py

```diff
@@ -37,6 +37,13 @@ def cli():
 
         WEB_SERVICE_URI = os.environ.get('WEB_SERVICE_URI')
 
+        FUNCX_S3_BUCKET_NAME = os.environ.get('FUNCX_S3_BUCKET_NAME')
+
+        if not FUNCX_S3_BUCKET_NAME:
+            raise Exception("S3 Storage bucket is required. Please specify by setting env variable `FUNCX_S3_BUCKET_NAME`")
```

Comment:
I don't believe we should be making it a hard failure if the env var is not set. That makes this a change that we cannot release without changes in helm and changes to AWS configuration for the service being finished. It effectively breaks `main` until all of the infrastructure changes are done.

I [raised this on the funcx-forwarder PR](https://github.com/funcx-faas/funcx-forwarder/pull/45#discussion_r750474383) as well.

I think that the correct behavior if the env var is not set is to failover to the previous behavior (writing to `task.result`).

All this requires is that the use of `RedisS3Storage` is abstracted behind a method or set of methods for getting and/or setting results. Like the web service, this should only need to read the result.

As an example, consider this `get_task_result` method (from [my variant of this for the web service](https://github.com/funcx-faas/funcx-web-service/pull/278)):
https://github.com/funcx-faas/funcx-web-service/blob/e7ca2743e8ecefb8b224c579918f780bf0d12052/funcx_web_service/task_storage.py#L63-L68